File: TutorialGame/TutorialGame/GameLoop/GameTime.cs
using System;

namespace TutorialGame
{
    public class GameTime
    {
        private float _deltaTime = 0f;
        private float _timeScale = 1f;

        public float TimeScale
        {
            get { return _timeScale; }
            set { _timeScale = value; }
        }

        public float DeltaTime
        {
            get { return _deltaTime * _timeScale; }
            private set { _deltaTime = value; }
        }

        public float DeltaTimeUnscaled
        {
            get { return _deltaTime; }
        }

        public float TotalTimeElapsed
        {
            get;
            private set;
        }

        public GameTime()
        {
        }

        public void Update(float deltaTime, float totalTimeElapsed)
        {
            _deltaTime = deltaTime;
            TotalTimeElapsed = totalTimeElapsed;
        }

    }
}

File: TutorialGameVB/TutorialGameVB/GameLoop/GameTime.vb
Imports System

Namespace TutorialGame

    Public Class GameTime

        Private _deltaTime As Single = 0F

        Private _timeScale As Single = 1.0F

        Public Property TimeScale As Single
            Get
                Return _timeScale
            End Get

            Set(ByVal value As Single)
                _timeScale = value
            End Set
        End Property

        Public Property DeltaTime As Single
            Get
                Return _deltaTime * _timeScale
            End Get

            Private Set(ByVal value As Single)
                _deltaTime = value
            End Set
        End Property

        Public ReadOnly Property DeltaTimeUnscaled As Single
            Get
                Return _deltaTime
            End Get
        End Property

        Public Property TotalTimeElapsed As Single

        Public Sub New()
        End Sub

        Public Sub Update(ByVal deltaTime As Single, ByVal totalTimeElapsed As Single)
            _deltaTime = deltaTime
            totalTimeElapsed = totalTimeElapsed
        End Sub
    End Class

End Namespace